From d08e54b68ca375e361e5cd5bd0e82ed3cd6033e3 Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
Date: Fri, 5 Mar 2021 15:00:03 -0800
Subject: [PATCH] main.conf: Enable passing false to DeviceID

This adds support for setting DeviceID to false so plaforms can disable
DeviceID.

Fixes: https://github.com/bluez/bluez/issues/101

Author: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Upstream-status: Backport
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 src/main.c    | 8 +++++++-
 src/main.conf | 2 +-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index b66e2b8cb..572dc939c 100644
--- a/src/main.c
+++ b/src/main.c
@@ -188,10 +188,16 @@ static void parse_did(const char *did)
 	int result;
 	uint16_t vendor, product, version , source;
 
-	/* version and source are optional */
+	vendor = 0x0000;
+	product = 0x0000;
 	version = 0x0000;
 	source = 0x0002;
 
+	if (!strcasecmp(did, "false")) {
+		source = 0x0000;
+		goto done;
+	}
+
 	result = sscanf(did, "bluetooth:%4hx:%4hx:%4hx",
 					&vendor, &product, &version);
 	if (result != EOF && result >= 2) {
diff --git a/src/main.conf b/src/main.conf
index ad36638b7..f47cab46d 100644
--- a/src/main.conf
+++ b/src/main.conf
@@ -26,7 +26,7 @@
 # Use vendor id source (assigner), vendor, product and version information for
 # DID profile support. The values are separated by ":" and assigner, VID, PID
 # and version.
-# Possible vendor id source values: bluetooth, usb (defaults to usb)
+# Possible vendor id source values: bluetooth, usb (default) or false (disabled)
 #DeviceID = bluetooth:1234:5678:abcd
 
 # Do reverse service discovery for previously unknown devices that connect to

